apps/fluent-tester/src/E2E/pages/ButtonTestPage.ts

```diff
@@ -0,0 +1,14 @@
+import { BUTTON_TESTPAGE } from '../../RNTester/Consts';
+import { By } from './BootTestPage';
+
+class ButtonTestPage {
+  isPageLoaded(): boolean {
```

Comment:
If you wanted to you could factor this out into a base class and extend that class. Something like
```
isPageLoaded(): boolean {
  return this._testPage.isDisplayed();
}
```
Then each test page only needs to implement `get _testPage()`.

Reply:
How would this work with a base class? If I make the base class and create the function above, _testPage is not a valid variable because the basePage class doesn't have a _testPage. Or am I misunderstanding something?

Reply:
Why isn't _testPage a valid variable? To be clear, base needs some implementation of `get _testPage`, it can return `By(dummy)`. The derived classes just need to override `get _testPage`. Hopefully, that's a little clearer.

Reply:
Yup I get it now. Just updated in the newest commit. Thank you

Reply:
Let me know if this was what you envisioned

Reply:
Yeah that's what I was imagining 👍. You could even make the base throw an error in `get _testPage` so that no one ever tries to use it.